modules/core/src/main/scala/fs2/kafka/KafkaProducer.scala

```diff
@@ -56,6 +54,10 @@ abstract class KafkaProducer[F[_], K, V] {
   def produce[P](
     records: ProducerRecords[P, K, V]
   ): F[F[ProducerResult[P, K, V]]]
+
+  def partitionsFor(
+    topic: String,
+  ): F[List[PartitionInfo]]
```

Comment:
This needs to be removed here so it's only defined in `KakfaProducer.PartitionFor` - I'm going to take the liberty of doing so.
```suggestion
```